### PR TITLE
core(network): handle invalid NetworkRequests

### PR DESCRIPTION
**Pull request by exterkamp**
**Summary**
Adds `isValid` property to a NetworkRequest and ignores invalid requests when making the list of records to surface to rest of application.

**Related Issues/PRs**
fixes: #6145

### CODE REVIEW

File: lighthouse-core/lib/network-recorder.js [hunk 1]
```diff
@@ -339,8 +339,8 @@ class NetworkRecorder extends EventEmitter {
     // playback all the devtools messages to recreate network records
     devtoolsLog.forEach(message => networkRecorder.dispatch(message));
 
-    // get out the list of records
-    const records = networkRecorder.getRecords();
+    // get out the list of records & filter out invalid records
+    const records = networkRecorder.getRecords().filter(record => record.isValid);
```
Comment (patrickhulce) [inline lighthouse-core/lib/network-recorder.js]: you might want to add `if (!request || !request.isValid)` to line 323 above too, behavior of the other `on*` methods is unexpected if all the properties are unset

Comment (exterkamp) [reply lighthouse-core/lib/network-recorder.js]: Good call.  Fixed!

File: lighthouse-core/lib/network-request.js [hunk 2]
```diff
@@ -125,27 +126,31 @@ module.exports = class NetworkRequest {
...
-    this.requestMethod = data.request.method;
-
-    this.initiator = data.initiator;
-    this.resourceType = data.type && RESOURCE_TYPES[data.type];
-    this.priority = data.request.initialPriority;
-
-    this.frameId = data.frameId;
-    this.isLinkPreload = data.initiator.type === 'preload' || !!data.request.isLinkPreload;
+    try {
+      // try to construct the url and fill in request
```
Comment (brendankenny) [inline lighthouse-core/lib/network-request.js]: Should we unconditionally catch all errors in here? Admittedly there's not anything else in here that will throw if our type checking is correct, but we might add more functionality in the future and any errors will fail silently.

What do you think about limiting the try/catch to the url constructor?

Like
```js
let url;
try {
  // try to construct the url and fill in request
  url = new URL(data.request.url);
} catch (e) {
  return; // isValid left false, all other data is blank
}
this.isValid = true;
this.url = data.request.url;
// ...
```

Comment (exterkamp) [reply lighthouse-core/lib/network-request.js]: Yup that was my first instinct!  Reverted.

**Comment by brendankenny** [comment]
@exterkamp should we add this to our smoke tests? Would it be easy to add the @font-face thing to https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-cli/test/fixtures/dobetterweb/dbw_tester.html? The expectations shouldn't need to be updated, the test would just be that the run completes and the results still come back as expected before.

**Comment by brendankenny** [comment]
we'll punt on the smoke test to a future change (pending getting it to work in a non-complicated way :)

**Pull request merged**